### DISCUSSION TITLE
[FEATURE]1panel-network默认是没有开启IPV6的，如何在不删除的情况下支持IPV6

### DISCUSSION THREAD

**Discussion by WudaxStudio** [1Panel-dev/1Panel#6766] [Q&A]
### 1Panel 版本

v1.10.19-lts

### 请描述您的需求或者改进建议

自带的1panel-network是默认支持IPV4的，但是并没有开启IPV6的功能，目前无法像创建那样，可以进行可视化的管理和配置

### 请描述你建议的实现方案

自带的1panel-network是默认支持IPV4的，但是并没有开启IPV6的功能，目前无法像创建那样，可以进行可视化的管理和配置，只能删除，然后重新添加一个新的，但是这样可能会导致很多容器出现问题，有什么办法可以在不删除1panel-network的网络情况下，修改其支持IPV6，或者可视化直接启用或修改IPV6的功能。像下图一样，修改也是这个页面，便于操作和更新
![image](https://github.com/user-attachments/assets/9cb38d95-5c0c-43d3-8cfa-e9832ee9b13d)

**Comment by wanghe-fit2cloud** [comment]
由于创建 IPv6 容器网络时需要指定子网、网关等信息，所以 1Panel 默认只支持 IPv4。如果要启用 IPv6，必须进行二次修改。
而 Docker 默认不支持直接修改已创建网络的配置，因此需要先删除网络再重新创建。

如果大家有更好的解决办法，欢迎在此继续讨论。

**Reply by feewg** [reply]
安装1panel的时候能不能检测环境，如果存在v6的话就自动配置
或者说安装1panel的时候默认不要安装docker，进入面板后再提示安装，然后给一些安装选项

**Reply by WudaxStudio** [reply]
是的，理论上应该可以获取当前网络IPV6的信息

**Reply by wanghe-fit2cloud** [reply]
主要是需要配置正确的子网和网关，这是创建 IPv6 容器网络的关键。
```
docker network create --ipv6 --subnet <IPv6子网范围> --gateway <IPv6网关> 1panel-network
```

**Comment by wanghe-fit2cloud** [comment]
论坛文章：https://bbs.fit2cloud.com/t/topic/8590/4/

**Reply by feewg** [reply]
对于服务器来说似乎没有这个问题，大部分服务器都是有固定的v6地址或者固定的范围的

**Reply by WudaxStudio** [reply]
1PANEL其实使用的大部分用户都是家用宽带用户的，所以这个需求大部分是需要的哦
